File: frontend/src/camera-list/camera-data.ts
import { OutputType } from './output-type';
import { SourceType } from './source-type';

export type CameraData = {
    readonly id: string
    readonly sourceType: SourceType
    readonly outputType: OutputType
    readonly name: string
    readonly ip: string
    readonly port: number
    readonly destPort: number
    readonly fileName: string
    readonly running: boolean
}

File: cameradata.go
package main

type CameraData struct {
	Id         string     `json:"id"`
	SourceType SourceType `json:"sourceType"`
	OutputType OutputType `json:"outputType"`
	Name       string     `json:"name"`
	Ip         string     `json:"ip"`
	Port       int        `json:"port"`
	DestPort   int        `json:"destPort"`
	FileName   string     `json:"fileName"`
	Running    bool       `json:"running"`
}
